node/node.go

```diff
@@ -938,14 +938,31 @@ func initSectorBuilderForNode(ctx context.Context, node *Node, proofsMode types.
 	// metadata in the staging directory, it should be in its own directory.
 	//
 	// Tracked here: https://github.com/filecoin-project/rust-fil-proofs/issues/402
-	sectorDir := paths.GetSectorPath(node.Repo.Config().SectorBase.RootDir)
+	repoPath, err := node.Repo.Path()
```

Comment:
I don't love adding this method to the repo interface.  It is the cleanest way to capture repo path info passed with the `--repodir` flag so that the sectors path can default to `repodir/../filecoin_sectors`

Reply:
Looks like we had [similar ideas](https://github.com/filecoin-project/go-filecoin/pull/2663/files#diff-1bae8ed1271a68f399d2ee0151eb38f6R496) here, although I only added it to `FSRepo` and kept it off the interface.  I did it this way because it is my understanding that a "repo" is an abstract concept, and therefore may not always have a path (e.g. Memrepo). What do you think?

I am flagging this because I rank merging the `inspect` command as lower priority that this MFC business. If it makes things easier I will wait on merging the `inspect` code until this goes through, what do you prefer?

Reply:
Your way is better in terms of clean abstractions but the node only has access to the interface and must be able to init a sector builder when initialized with a mem repo for test so this addition to the interface is needed without other intervention.

I would prefer merging this first, but the merge conflict with the inspect command should be easy to resolve.

Reply:
If we are only doing this for the sake of testing then why not just type assert ([example](https://github.com/filecoin-project/go-filecoin/pull/2663/files#diff-b989720bf97c51e8761fd9483d5cedd6R82))? I don't mean to be an Armchair Expect here, I haven't read this PR in full -- this bit just caught my attention while skimming so take these comments as non-blocking.

You are good to merge from my end, the conflicts in `inspect` will be simple.

Reply:
Is this only to handle the sector dir being empty in the config? I think we should disallow that, and require it to be set in the config. `init` should set a default of `.filecoin_sectors` (where the `.filecoin` prefix is taken from the basename of the --repodir value), which can be overridden by a flag specifying some other path (or just changed in the file afterwards).

It's just mandatory configuration, with a default. I think this will remove some ugliness in the IPTB plugin too.

Note that I'm touching `init` soon to set up the symlink, so could roll this in to #2684.